lib/mongo/collection/view/writable.rb

```diff
@@ -317,8 +321,10 @@ def update_many(spec, opts = {})
         def update_one(spec, opts = {})
           update_doc = { Operation::Q => filter,
                          Operation::U => spec,
-                         Operation::MULTI => false,
-                         Operation::UPSERT => !!opts[:upsert] }
+                         }
+          if opts[:upsert]
+            update_doc[:upsert] = true
```

Comment:
I noticed upon upgrading to 2.11.2 that our software's uses of `upsert: true` failed to insert new documents, and traced it to this change. Replacing the `:upsert` symbol with `Operation::UPSERT` (which is a _string_) fixed our problem. We've downgraded to v2.9.x in the meantime.

Reply:
@joeyAghion Can you please create a new jira ticket (https://jira.mongodb.org/browse/RUBY) describing your issue, including the following:

- a code snippet reproducing the problem
- what you expected to happen
- what actually happened

Reply:
https://jira.mongodb.org/browse/RUBY-2054